lib/base58encode.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Base58Encode do
+  @moduledoc """
+  Provides the encode function which takes a binary and return it's
+  reprentation in base58
+  """
+
+  @doc """
+  ## Examples
+
+    iex> Base58Encode.encode("hello")
+    "Cn8eVZg"
+
+    iex> Base58Encode.encode(42)
+    :error
+  """
+
+ def encode(binary) when is_binary(binary) do
+   case Integer.parse(Base.encode16(binary), 16) do
```

Comment:
Not very clear to me why you `Base.encode16` the binary and then change `Integer.parse` it back.

Reply:
agree with you that this step as a hacky feeling.
The idea is to first encode the String/binary to hexadecimal:
"foo" becomes "666F6F".
Then from the hexadecimal we can call the `Integer.parse` function to convert the value to decimal, so "666F6F" becomes 6713199
And now we can encode this integer to base58 by using the modulo operator with 58.
I haven't find an easy way to avoid creating the hexadecimal and I don't think we can deduce directly the integer from the binary, ie <<102, 111, 111>> to 6713199 but let me know if you find a better way to do this :+1:

Reply:
Ahhh that makes a lot of sense. Yeah I don't know a way to do this either (right now). I'll try and look into this but it is in no way a blocker. Thanks

Reply:
Asked a [SO](https://stackoverflow.com/questions/54441543/how-to-convert-a-binary-to-a-base10-decimal-integer-in-elixir) question on this.

Reply:
up-voted. ⬆️

Reply:
Good answer:
![image](https://user-images.githubusercontent.com/6057298/51989674-6edeb580-249f-11e9-80d0-439216fb97a5.png)